app/javascript/controllers/unassign_evaluator_submission_modal_controller.js

```diff
@@ -38,30 +37,29 @@ export default class extends Controller {
   }
 
   setValues(dataset) {
-    this.submissionIdValue = dataset.submissionId;
-    this.evaluatorIdValue = dataset.evaluatorId;
+    this.assignmentIdValue = dataset.assignmentId;
     this.phaseIdValue = dataset.phaseId;
   }
 
   unassignEvaluatorSubmission() {
     const csrfToken = document.querySelector('meta[name="csrf-token"]').content;
     
-    fetch(`/phases/${this.phaseIdValue}/evaluator_submission_assignments?evaluator_id=${this.evaluatorIdValue}`, {
+    fetch(`/phases/${this.phaseIdValue}/evaluator_submission_assignments/${this.assignmentIdValue}`, {
       method: 'PATCH',
       headers: {
         'Content-Type': 'application/json',
         'X-CSRF-Token': csrfToken,
         'Accept': 'application/json'
       },
       body: JSON.stringify({
-        submission_id: this.submissionIdValue,
         status: 'unassigned'
       })
     })
     .then(response => response.json())
     .then(data => {
       if (data.success) {
-        window.location.href = `/phases/${this.phaseIdValue}/evaluator_submission_assignments?evaluator_id=${this.evaluatorIdValue}`;
+        const evaluatorId = new URLSearchParams(window.location.search).get('evaluator_id');
+        window.location.href = `/phases/${this.phaseIdValue}/evaluator_submission_assignments?evaluator_id=${evaluatorId}`;
```

Comment:
this is essential doing `location.reload()` right? I have a different idea, what if you redirect to the path provided in the response data. you could render the url to `data.redirect_to` and then reuse that same redirect for both success and error since the backend will determine where to send the user. I think this will also display the flash message that was stored after the browser redirects (so you may not need the `alert()` either. if you want you can still log a console.error() for developer debugging help messages.
```suggestion
        window.location.href = data.redirect_to;
```